rholang/src/test/scala/coop/rchain/rholang/PeekSpec.scala

```diff
@@ -0,0 +1,42 @@
+package coop.rchain.rholang
+
+import org.scalatest._
+
+import coop.rchain.metrics.Metrics
+import coop.rchain.models.Expr.ExprInstance.{GInt, GString}
+import coop.rchain.models.rholang.implicits._
+import coop.rchain.shared.Log
+import coop.rchain.rholang.interpreter.{EvaluateResult, InterpreterUtil, Runtime}
+
+import monix.eval.Task
+import monix.execution.Scheduler.Implicits.global
+
+import scala.concurrent.duration._
+
+class PeekSpec extends FlatSpec with Matchers {
+
+  import Resources._
+
+  implicit val logF: Log[Task]            = new Log.NOPLog[Task]
+  implicit val noopMetrics: Metrics[Task] = new Metrics.MetricsNOP[Task]
+
+  val tmpPrefix = "peek-spec-"
+
+  "peek" should "not remove read data" in {
+    mkRuntime[Task](tmpPrefix)
+      .use { runtime =>
+        implicit val c = runtime.cost
+        for {
+          res1 <- InterpreterUtil.evaluate[Task](runtime, """@1!("v1") | for(_ <<- @1) { Nil }""")
+          _    = res1.errors shouldBe empty
+          res2 <- InterpreterUtil.evaluate[Task](runtime, """for(_ <- @1) { @2!("v2") }""")
```

Comment:
It makes sense to add a similar test for persistent sends. One that asserts that peek does not duplicate v1 at @1.